src/vs/workbench/services/files/node/fileService.ts

```diff
@@ -221,7 +221,14 @@ export class FileService implements files.IFileService {
 			if (exists) {
 				createParentsPromise = Promise.as(null);
 			} else {
-				createParentsPromise = pfs.mkdirp(paths.dirname(absolutePath));
+				//Do not try to make parent directory if saving to drive root
+				var dirName = paths.dirname(absolutePath);
+				if(dirName.slice(-2) != ':\\') {
```

Comment:
So the problem is that a recursive `mkdir` is always attempted if the file does not yet exists? How about doing a `pfs.stat` on `paths.dirname(absolutePath)` and see if it exists? Root directories should return true to that and it should be fine cross-platform too.